lib/src/main/java/com/diffplug/spotless/java/FormatAnnotationsStep.java

```diff
@@ -0,0 +1,474 @@
+					"NTDSide",
+					"NTDTop",
+					"Nullable",
+					"NullableType",
+					"Odd",
+					"OptionalBottom",
+					"OrderNonDet",
+					"Parent",
+					"PatternA",
+					"PatternAB",
+					"PatternAC",
+					"PatternB",
+					"PatternBC",
+					"PatternBottomFull",
+					"PatternBottomPartial",
+					"PatternC",
+					"PatternUnknown",
+					"Poly",
+					"PolyAll",
+					"PolyConstant",
+					"PolyDet",
+					"PolyEncrypted",
+					"PolyFenum",
+					"PolyIndex",
+					"PolyInitializedFields",
+					"PolyInterned",
+					"PolyKeyFor",
+					"PolyLength",
+					"PolyLowerBound",
+					"PolyMustCall",
+					"PolyNull",
+					"PolyNullType",
+					"PolyPresent",
+					"PolyRaw",
+					"PolyReflection",
+					"PolyRegex",
+					"PolySameLen",
+					"PolySignature",
+					"PolySigned",
+					"PolyTainted",
+					"PolyTestAccumulation",
+					"PolyTypeDeclDefault",
+					"PolyUI",
+					"PolyUnit",
+					"PolyUpperBound",
+					"PolyValue",
+					"PolyVariableNameDefault",
+					"Positive",
+					"Present",
+					"PrimitiveType",
+					"PropertyKey",
+					"PropertyKeyBottom",
+					"PurityUnqualified",
+					"Qualifier",
+					"radians",
+					"Raw",
+					"ReflectBottom",
+					"Regex",
+					"RegexBottom",
+					"RegexNNGroups",
+					"ReportUnqualified",
+					"s",
+					"SameLen",
+					"SameLenBottom",
+					"SameLenUnknown",
+					"SearchIndexBottom",
+					"SearchIndexFor",
+					"SearchIndexUnknown",
+					"Sibling1",
+					"Sibling2",
+					"SiblingWithFields",
+					"SignatureBottom",
+					"Signed",
+					"SignednessBottom",
+					"SignednessGlb",
+					"SignedPositive",
+					"SignedPositiveFromUnsigned",
+					"Speed",
+					"StringVal",
+					"SubQual",
+					"Substance",
+					"SubstringIndexBottom",
+					"SubstringIndexFor",
+					"SubstringIndexUnknown",
+					"SuperQual",
+					"SwingBoxOrientation",
+					"SwingCompassDirection",
+					"SwingElementOrientation",
+					"SwingHorizontalOrientation",
+					"SwingSplitPaneOrientation",
+					"SwingTextOrientation",
+					"SwingTitleJustification",
+					"SwingTitlePosition",
+					"SwingVerticalOrientation",
+					"t",
+					"Tainted",
+					"Temperature",
+					"TestAccumulation",
+					"TestAccumulationBottom",
+					"TestAccumulationPredicate",
+					"This",
+					"Time",
+					"Top",
+					"TypeDeclDefaultBottom",
+					"TypeDeclDefaultMiddle",
+					"TypeDeclDefaultTop",
+					"UbTop",
+					"UB_TOP",
+					"UI",
+					"UnderInitialization",
+					"Unique",
+					"UnitsBottom",
+					"UnknownClass",
+					"UnknownCompilerMessageKey",
+					"UnknownFormat",
+					"UnknownInitialization",
+					"UnknownInterned",
+					"UnknownKeyFor",
+					"UnknownLocalizableKey",
+					"UnknownLocalized",
+					"UnknownMethod",
+					"UnknownPropertyKey",
+					"UnknownRegex",
+					"UnknownSignedness",
+					"UnknownThis",
+					"UnknownUnits",
+					"UnknownVal",
+					"Unsigned",
+					"Untainted",
+					"UpperBoundBottom",
+					"UpperBoundLiteral",
+					"UpperBoundUnknown",
+					"Value",
```

Comment:
As per [the docs](https://github.com/diffplug/spotless/tree/main/plugin-gradle#formatannotations), you can use `addTypeAnnotation` and `removeTypeAnnotation` to tweak for your case.

Reply:
It's referring to `lombok.Value`.  I agree that fully-qualified names are the way to go.  This would be trivial in (say) google-java-format -- the mechanism already exists -- but would require much more work within Spotless.

Reply:
https://github.com/diffplug/spotless/pull/1367 changes `@Value` from a type annotation.

Reply:
Thanks for the clarification!

As you mention google-java-format: Have you thought about fixing this issue in the fork palantir-java-format? They seem to be more open for community contributions than Google.

Reply:
> As you mention google-java-format: Have you thought about fixing this issue in the fork palantir-java-format? They seem to be more open for community contributions than Google.

This is a good idea.

Is it actively maintained?  I see only 3 substantive [commits](https://github.com/palantir/palantir-java-format/commits/develop) since April, and it has not integrated recent changes (including new features and bug commits) from google-java-format.  An example is https://github.com/google/google-java-format/commit/865cff01 from August 2021, which would be necessary.

Reply:
It is well-enough maintained for my decision to use it. :-) (I needed an alternative to google-java-format that doesn't force me to use 100 chars lines and 2-spacing-indentation.)

In January, they cherry-picked several commits from upstream, so I assume they are inclined to keep up with google-java-format.